### ISSUE TITLE
JsonView support

### ISSUE THREAD

**Issue by monoc44** [swagger-api/swagger-core#2079]
Dear team,

JSON views are a very convenient and powerful way to not duplicate models used by resources. Many developers already rose this point and believe Swagger should support it. However it seems that all the requests to support this feature have been turned down.

What about generating a single and individual definition entry when @JsonView is detected?

For instance, taking this example below (pseudo code):

```
@GET(/serviceA)
@JsonView(ViewA.class)
public MyModel methodA() {...}

@GET(/serviceB)
@JsonView(ViewB.class)
public MyModel methodB() {...}
```

```
class MyModel {

  Integer id;

  @JsonView(ViewA.class)
  String propertyForServiceA;

  @JsonView(ViewA.class)
  String propertyNotForServiceB;

  @JsonView(ViewB.class)
  String propertyForServiceB;
  ...
}
```

=> parsing these two resources will translate today into the following OAI JSON file:

```
paths: {
   /serviceA:
     get ->  responses: { 200-> $ref: #/definitions/MyModel }
   /serviceB: {
     get ->  responses: { 200-> $ref: #/definitions/MyModel }
},
definitions: {
  MyModel:
      type: object,
      properties: [
        id,
        propertyForServiceA,
        propertyNotForServiceB,
        propertyForServiceB
      ]
}
```

Sadly, only one definition model will be generated after Swagger scans either serviceA or serviceB.

Would it be difficult to instead generate a 'contextualized' definition model associated to each individual service, hence 'supporting' JsonViews?

The generated OAI file could then look something like:

```
paths: {
   /serviceA:
     get ->  responses: { 200-> $ref: #/definitions/MyModel.serviceA.methodA}
   /serviceB: {
     get ->  responses: { 200-> $ref: #/definitions/MyModel.serviceB.methodB}
},
definitions: {
  MyModel.serviceA.methodA:
      className: xxx.xxx.MyModel
      type: object,
      properties: [
        id,
        propertyForServiceA,
        propertyNotForServiceB
      ]
  MyModel.serviceB.methodB:
      className: xxx.xxx.MyModel      
      type: object,
      properties: [
        id,
        propertyForServiceB
      ]
}
```

Please let me know what you think. Thanks.

**Comment by fehguy** [comment]
Hi, a bit of confusion.  There's no philosophical reason why they're not supported--they're "just not supported" so sending a PR would be just fine.  But we do need to adhere to the specification so things like `className` are illegal as you're proposing.

I would need to think more about what you're actually proposing but please don't get me wrong--there's nothing wrong with `@JsonView` other than the fact that it's more work, and any implementation of it inside swagger-core needs to adhere to the spec.

**Comment by monoc44** [comment]
That's a great news and I can create a pull request if needed. Please let me know what I should put inside though.

About `className`, I agree that it will be something to submit to OAI. Anyway, this information is not that important and can be omitted easily.

Thanks for taking a look, much appreciated!

**Comment by frantuma** [comment]
implemented in #2662 and #2681
